framework/src/org/checkerframework/framework/util/element/ElementAnnotationUtil.java

```diff
@@ -87,7 +87,17 @@ public static void applyAllElementAnnotations(
     static void addAnnotationsFromElement(
             final AnnotatedTypeMirror type, final List<? extends AnnotationMirror> annotations) {
         AnnotatedTypeMirror innerType = AnnotatedTypes.innerMostType(type);
-        innerType.addAnnotations(annotations);
+        if (innerType != type) {
+            for (AnnotationMirror annotation : annotations) {
+                if (innerType.addAnnotation(annotation)) {
+                    ErrorReporter.warnOnce(
```

Comment:
@mernst I would appreciate feed back on this warning message.

Reply:
Looks good to me.